haystack/nodes/preprocessor/preprocessor.py

```diff
@@ -232,7 +237,7 @@ def clean(
         text = document.content
         if clean_header_footer:
             text = self._find_and_remove_header_footer(
-                text, n_chars=300, n_first_pages_to_ignore=1, n_last_pages_to_ignore=1
+                str(text), n_chars=300, n_first_pages_to_ignore=1, n_last_pages_to_ignore=1
```

Comment:
Great, I'll double-check the text variable. I tried to minimize code changes on this. I just noticed the string and binary issues because of the linter warns. Then I noticed it could be string or DataFrame. I'll include pre-check on text.